src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -467,6 +467,62 @@ namespace cryptonote
     return true;
   }
   //---------------------------------------------------------------
+  bool add_serializable_tx_extra_field_to_tx_extra(std::vector<uint8_t>& tx_extra, tx_extra_field& field, uint8_t tag)
+  {
+    std::ostringstream oss;
+    binary_archive<true> ar(oss);
+    bool r = ::do_serialize(ar, field);
+    if (!r) return false;
+
+    const size_t pos = tx_extra.size();
+    const std::string tx_extra_str = oss.str();
+    tx_extra.resize(tx_extra.size() + 1 + tx_extra_str.size());
+    tx_extra[pos] = tag;
+    memcpy(&tx_extra[pos+1], tx_extra_str.data(), tx_extra_str.size());
+
+    return true;
+  }
+  //---------------------------------------------------------------
+  bool add_pub_spendkey_to_tx_extra(std::vector<uint8_t>& tx_extra, const crypto::public_key& tx_pub_key)
+  {
+    tx_extra.resize(tx_extra.size() + 1 + sizeof(crypto::public_key));
+    tx_extra[tx_extra.size() - 1 - sizeof(crypto::public_key)] = TX_EXTRA_TAG_PUB_SPENDKEY;
```

Comment:
I changed this in a later commit to be EXTRA_TAG_ACCOUNT_PUBLIC_ADDRESS, will squash commits and create a new PR